makefiles/common.mk

```
@@ -192,16 +202,17 @@ SHOULD_STRIP := $(call __theos_bool,$(or $(strip),$(STRIP),$(_THEOS_SHOULD_STRIP
 
 ifeq ($(SHOULD_STRIP),$(_THEOS_TRUE))
 OPTFLAG ?= -Os
-SWIFT_OPTFLAG ?= -O -whole-module-optimization
+SWIFT_OPTFLAG ?= -O -whole-module-optimization -num-threads 1
```

Is there a reason we want to run single threaded here?
```suggestion
SWIFT_OPTFLAG ?= -O -whole-module-optimization -num-threads $(shell $(_THEOS_PLATFORM_GET_LOGICAL_CORES))
```

---

iirc Xcode uses one thread for release builds — I'm not sure what the rationale is but I assume there's a build time vs runtime tradeoff here.

---

It's because of optimisations. If you compile single threaded, optimisations can be done on everything. If you compile multi-threaded, optimisations are done in each thread/compilation unit separately, so you miss out on opportunities to inline functions and eliminate dead code, because the threads don't know what the other threads are doing.